src/extraction/FStar.Extraction.ML.Term.fs

```diff
@@ -1288,6 +1288,8 @@ and term_as_mlexpr' (g:uenv) (top:term) : (mlexpr * e_tag * mlty) =
           begin
                match try_lookup_fv g fv with
                | None -> //it's been erased
+                 Errors.log_issue t.pos (Errors.Error_CallToErased,
+                                         BU.format1 "Attempting to extract a call into erased function %s" (Print.fv_to_string fv));
```

Comment:
I'm not sure this is detecting that `fv` has been not extracted correctly. For instance:


```

noextract
let f (x:int) = x + 1

let g x = f x
```

does not produce a warning and extracts to

```
let (g : Prims.int -> Prims.int) = fun uu___ -> (fun x -> Obj.magic ()) uu___
```

Reply:
Yeah, there's a second call to try_lookup_fv in the same file where we do some specialization to extract an application whose head symbol is an fvar. That one also needs to trap and log this issue.

Reply:
In fact, I don't think you should warn in either of these cases.

In case a top-level function is legitimately erased to unit, it is just dropped from the environment (e.g., a top-level lemma or a squash-returning function). And if that is then correctly extracted to the `()` literal. We shouldn't error out in those cases.

We might need to add more metadata to the extraction environment to distinguish the cases of extracting a legitimately erased term from a noextract term.